SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestMultipleOwnership/SpatialTestMultipleOwnership.cpp

```diff
@@ -0,0 +1,213 @@
+		[this](float DeltaTime) {
+			TArray<AActor*> FoundPawns;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AMultipleOwnershipPawn::StaticClass(), FoundPawns);
+
+			// Make sure the correct number of MultipleOwnershipPawns is visible
+			if (FoundPawns.Num() != 2)
+			{
+				return;
+			}
+
+			// Preemptively empty the Pawns array so that we don't get any issues with index-based addressing later on in the test
+			MultipleOwnershipPawns.Empty();
+
+			for (AActor* Pawn : FoundPawns)
+			{
+				AMultipleOwnershipPawn* MultipleOwnershipPawn = Cast<AMultipleOwnershipPawn>(Pawn);
+				if (MultipleOwnershipPawn != nullptr)
+				{
+					MultipleOwnershipPawns.Add(MultipleOwnershipPawn);
+				}
+			}
+			MultipleOwnershipPawns.Sort([](AMultipleOwnershipPawn& LHS, AMultipleOwnershipPawn& RHS) -> bool {
+				return LHS.GetActorLocation().Y < RHS.GetActorLocation().Y;
+			});
+			// Double checking that we have the correct number of MultipleOwnershipPawns, just to make sure that the casting above was
+			// successful and no issues arise from that
+			if (MultipleOwnershipPawns.Num() == 2)
+			{
+				FinishStep();
+			}
+		},
+		5.0f);
+
+	// Step definition for Client 1 to send a Server RPC which is used multiple times during the test. RPCs are expected to miss twice on
+	// the first pass, then once, then 0 times..
+	FSpatialFunctionalTestStepDefinition ClientSendRPCStepDefinition;
+	ClientSendRPCStepDefinition.StepName = TEXT("ClientSendRPCsTest");
+	ClientSendRPCStepDefinition.bIsNativeDefinition = true;
+	ClientSendRPCStepDefinition.NativeStartEvent.BindLambda([this]() {
+		for (AMultipleOwnershipPawn* MultipleOwnershipPawn : MultipleOwnershipPawns)
+		{
+			MultipleOwnershipPawn->ServerSendRPC();
+		}
+
+		FinishStep();
+	});
+
+	// Client 1 sends an RPC from both MultipleOwnershipPawns
+	AddStepFromDefinition(ClientSendRPCStepDefinition, FWorkerDefinition::Client(1));
+
+	// All workers check that both MultipleOwnershipPawns have received 0 RPCs, therefore that the RPCs were ignored since the
+	// MultipleOwnershipPawns had no owner.
+	AddStep(
+		TEXT("SpatialTestMultipleOwnershipAllWorkersCheckCorrectRPCs"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			if (MultipleOwnershipPawns[0]->ReceivedRPCs == 0 && MultipleOwnershipPawns[1]->ReceivedRPCs == 0)
+			{
+				FinishStep();
+			}
+		},
+		10.0f);
+
+	// The server makes Client 1's PlayerController possess the first MultipleOwnershipPawn.
+	AddStep(TEXT("SpatialTestMultipleOwnershipPossessPawn1"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		APlayerController* PlayerController =
+			Cast<APlayerController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		PlayerController->Possess(MultipleOwnershipPawns[0]);
+
+		FinishStep();
+	});
+
+	// Client 1 sends an RPC from both MultipleOwnershipPawns
+	AddStepFromDefinition(ClientSendRPCStepDefinition, FWorkerDefinition::Client(1));
```

Comment:
I wonder if Client 1 could start this step before receiving updated possession data and log 2 errors.